core/src/main/java/software/amazon/awssdk/runtime/StandardMemberCopier.java

```diff
@@ -74,6 +74,19 @@ public static ByteBuffer copy(ByteBuffer bb) {
         if (bb == null) {
             return null;
         }
-        return bb.duplicate();
+
+        ByteBuffer copy = bb.isDirect() ? ByteBuffer.allocateDirect(bb.capacity()) : ByteBuffer.allocate(bb.capacity());
+
+        // Copy content
+        ByteBuffer toCopy = bb.asReadOnlyBuffer();
+        toCopy.rewind();
```

Comment:
I read that case, actually. There would be a performance hit for those partial-read cases for sure, depending on how much data is copied that probably won't be used. That said, zero people will be surprised by this implementation, and there's a chance of surprise from only performing a partial copy. I don't feel super strongly about it, though. I can change it if you feel like that risk avoidance isn't worth the potential cost.